lib/widgets/home.dart

```diff
@@ -307,29 +308,106 @@ void _showMainMenu(BuildContext context, {
-                child: SingleChildScrollView(
-                  padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 8),
-                  child: Column(children: menuItems)))),
-              const Padding(
-                padding: EdgeInsets.symmetric(horizontal: 16),
-                child: AnimatedScaleOnTap(
-                  scaleEnd: 0.95,
-                  duration: Duration(milliseconds: 100),
-                  child: BottomSheetDismissButton(
-                    style: BottomSheetDismissButtonStyle.close))),
-            ])));
+        child: _MainMenu(menuItems: menuItems));
     });
 }
 
+/// The main-menu sheet.
+///
+/// Figma link:
+///   https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=143-10939&t=s7AS3nEgNgjyqHck-4
+class _MainMenu extends StatelessWidget {
+  const _MainMenu({
+    required this.menuItems,
+  });
+
+  final List<Widget> menuItems;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    return SafeArea(
+      minimum: const EdgeInsets.only(bottom: 8),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          _MainMenuHeader(),
+          Flexible(child: InsetShadowBox(
+            top: 8, bottom: 8,
+            color: designVariables.bgBotBar,
+            child: SingleChildScrollView(
+              padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 8),
+              child: Column(children: menuItems)))),
+          const Padding(
+            padding: EdgeInsets.symmetric(horizontal: 16),
+            child: AnimatedScaleOnTap(
+              scaleEnd: 0.95,
+              duration: Duration(milliseconds: 100),
+              child: BottomSheetDismissButton(
+                style: BottomSheetDismissButtonStyle.close))),
+        ]));
+  }
+}
+
+class _MainMenuHeader extends StatelessWidget {
+  const _MainMenuHeader();
+
+  void _onPressed(BuildContext context) {
+    Navigator.pop(context); // Close the main menu.
+    Navigator.push(context,
+      MaterialWidgetRoute(page: const ChooseAccountPage()));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+
+    final realmIconUrl = store.resolveRealmIconUrl();
+
+    final placeholder = ColoredBox(color: designVariables.avatarPlaceholderBg);
+    final logo = realmIconUrl != null
+      ? RealmContentNetworkImage(
+          realmIconUrl,
+          frameBuilder: (_, child, frame, _) {
+            if (frame == null) return placeholder;
+            return child;
+          },
+          errorBuilder: (_, e, st) {
+            assert(debugLog('$e\n$st')); // TODO(log)
+            return placeholder;
+          })
+      : placeholder;
```

Comment:
nit: put a short fallback like this right next to the error/exception condition that causes it